rust/kernel/net/mod.rs

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Network subsystem in rust for linux kernel.
+
+use core::mem;
+
+pub mod device;
+pub mod ethtool;
+pub mod netlink;
+pub mod rtnl;
+pub mod skbuff;
+
+#[doc(inline)]
+pub use macros::rtnl_link_ops;
+
+/// Determine if the Ethernet address is a multicast.
+///
+/// Return true if the address is a multicast address.
+/// By definition the broadcast address is also a multicast address.
+///
+/// # Parameters
+/// - `addr`: Pointer to a six-byte array containing the Ethernet address
+///
+/// # Safety
+/// `addr` has to be a valid address pointer
+#[cfg(CONFIG_HAVE_EFFICIENT_UNALIGNED_ACCESS)]
+pub unsafe fn is_multicast_ether_addr(addr: *const u8) -> bool {
+    // SAFETY: addr is a valid address pointer as above required, and has efficient unaligned access as KCONFIG stats
```

Comment:
Please use Markdown in comments as well, so that we are consistent (and so that they can be easily copied). It could also useful if someone has a source viewer that renders comments.

Also, avoid saying "as above", we can be more precise to say "By the safety precondition" or similar.